packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -107,10 +107,10 @@ export const createAddressTransactionsProvider = ({
                   const rollbackTransactions = localTransactions.filter(
                     ({ blockHeader: { blockNo } }) => blockNo >= sinceBlock
                   );
-                  logger.debug(
-                    `Chain history has diverged. ${rollbackTransactions.length} transactions have been rolled back`
-                  );
-                  from(rollbackTransactions).subscribe(rollback$);
+
+                  from(rollbackTransactions)
+                    .pipe(tap((tx) => logger.debug(`Transaction ${tx.id} was rolled back`)))
+                    .subscribe(rollback$);
```

Comment:
@mirceahasegan not related to changes in this PR, but this might be a bug: `from()` emits all items and *completes* which I think will also call `complete()` on the subject. Could you please verify?

Reply:
Good catch @mkazlauskas . It was indeed completing the rollback subject.
I've updated a unit test to catch this and fixed it. Thank you.
[8754449](https://github.com/input-output-hk/cardano-js-sdk/pull/379/commits/8754449574b70d31374e913c8728a1063b1a0376)